src/app/dashboard/components/line.component.ts

```diff
@@ -0,0 +1,276 @@
+
+
+  _dialog = inject(MatDialog);
+  #autoRefreshService = inject(AutoRefreshService);
+  #iconsService = inject(IconsService);
+  #taskGrpcService = inject(TasksGrpcService);
+  #tasksIndexService = inject(TasksIndexService); 
+  #dashboardIndexService = inject(DashboardIndexService); 
+
+  total: number;
+  loadTasksStatus = false;
+  data: StatusCount[] = [];
+  availableFiltersFields: any[] = [];
+
+  refresh: Subject<void> = new Subject<void>();
+  stopInterval: Subject<void> = new Subject<void>();
+  interval: Subject<number> = new Subject<number>();
+  subscriptions: Subscription = new Subscription();
+  interval$: Observable<number> = this.#autoRefreshService.createInterval(this.interval, this.stopInterval);
+
+  ngOnInit(): void {
+    this.loadTasksStatus = true; 
+    this.availableFiltersFields = this.#dashboardIndexService.availableFiltersFields;                              
+  }
+
+
+  ngAfterViewInit() {
+    const mergeSubscription = merge(this.refresh, this.interval$).pipe(
+      startWith(0),
+      tap(() => (this.loadTasksStatus = true)),
+      switchMap(() => this.#taskGrpcService.countByStatu$()),
+    ).subscribe((data) => {
+      if (!data.status) {
+        return;
+      }
+
+      this.data = data.status;
+      this.total = data.status.reduce((acc, curr) => acc + curr.count, 0);
+
+      this.loadTasksStatus = false;
+    });
+
+    this.subscriptions.add(mergeSubscription);
+  }
+
+  ngOnDestroy() {
+    this.subscriptions.unsubscribe();
+  }
+
+  getIcon(name: string): string {
+    return this.#iconsService.getIcon(name);
+  }
+
+  getPageIcon(name: Page): string {
+    return this.#iconsService.getPageIcon(name);
+  }
+
+  autoRefreshTooltip(): string {
+    return this.#autoRefreshService.autoRefreshTooltip(this.line.interval);
+  }
+
+  columnsLabels(): Record<TaskSummaryColumnKey, string> {
+    return this.#tasksIndexService.columnsLabels;
+  }
+  onRefresh() {
+    this.refresh.next();
+  }
+
+  onIntervalValueChange( value: number) {
+    this.line.interval = value;
+
+    if(value === 0) {
+      this.stopInterval.next();
+    } else {
+      this.interval.next(value);
+      this.refresh.next();
+    }
+
+    this.saveChange.emit();
+
+  }
+
+  onToggleGroupsHeader() {
+    this.line.hideGroupsHeader = !this.line.hideGroupsHeader; 
+    this.toggleGroupsHeaderChange.emit(); 
+      
+  }
+
+  onEditNameLine(value: string) {
+    const dialogRef: MatDialogRef<EditNameLineDialogComponent, string> = this._dialog.open(EditNameLineDialogComponent, {
+      data: {
+        name: value
+      }
+    }); 
+
+
+    dialogRef.afterClosed().subscribe((result) => {
+      if (!result || result.trim() === '') {
+        return;
+      }
+
+      this.line.name = result;
+      this.editNameLineDialogChange.emit(); 
+    });
+
+    
+  }
+
+  onManageGroupsDialog() {
+    const dialogRef = this._dialog.open(ManageGroupsDialogComponent, {
```

Comment:
```suggestion
    const dialogRef = this.#dialog.open(ManageGroupsDialogComponent, {
```

You can also provide types into the generic of `open<3 generics>`.